kroxylicious/src/main/java/io/kroxylicious/proxy/internal/filter/BrokerAddressFilter.java

```diff
@@ -99,11 +103,12 @@ private <T> void apply(KrpcFilterContext context, T broker, Function<T, Integer>
         portSetter.accept(broker, downstreamAddress.port());
     }
 
-    private void doReconcileThenForwardResponse(ResponseHeaderData header, ApiMessage data, KrpcFilterContext context, Map<Integer, HostPort> nodeMap) {
+    private CompletionStage<ResponseFilterResult> doReconcileThenForwardResponse(ResponseHeaderData header, ApiMessage data, KrpcFilterContext context,
+                                                                                 Map<Integer, HostPort> nodeMap) {
         // https://github.com/kroxylicious/kroxylicious/issues/351
         // Using synchronous approach for now
         reconciler.reconcile(virtualCluster, nodeMap).toCompletableFuture().join();
-        context.forwardResponse(header, data);
-        LOGGER.debug("Endpoint reconciliation complete for  virtual cluster {}", virtualCluster);
+        LOGGER.debug("Endpoint reconciliation complete for virtual cluster {}", virtualCluster);
+        return context.responseFilterResultBuilder().forward(header, data).completed();
```

Comment:
Am I right in saying in the future we want this to move to something like?

```suggestion
        var result = context.responseFilterResultBuilder().forward(header, data);
        reconciler.reconcile(virtualCluster, nodeMap).toCompletableFuture().whenComplete((r, t) -> {
            LOGGER.debug("Endpoint reconciliation complete for virtual cluster {}", virtualCluster);
            result.completed();
        }
        return result;
```

Or is the plan more like 
```suggestion
        return reconciler.reconcile(virtualCluster, nodeMap).toCompletableFuture().whenComplete((r, t) -> {
            LOGGER.debug("Endpoint reconciliation complete for virtual cluster {}", virtualCluster);
            return context.forwardResponse(header, data);
        }
```

Reply:
I think it was a lifecycle distinction I was trying to draw, but only thought of the second after writing the first.